### PR TITLE
feat: support misc chains

### PR DESCRIPTION
**Pull request by nicosantangelo**
Depends on https://github.com/decentraland/decentraland-connect/pull/7

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1036,18 +1036,23 @@ async function wrapProviderToEthers() {
...
 
-### Web3x Eth instance
+### Eth instance
 
-Get web3x Eth instance as a legacy provider
+Get an Eth instance with your lib of choice
 
 ```ts
-import { createETH } from 'decentraland-dapps/dist/lib/eth'
+import { Eth } from 'web3x-es/eth'
```
Comment (nachomazzara) [inline README.md:1044]: May we need to add `import ethers from 'ethers'` following the example, right?

File: src/components/Intercom/IntercomWidget.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { isMobile, insertScript } from '../../lib/utils'
 
 import { IntercomWindow, IntercomSettings } from './Intercom.types'
 
-const intercomWindow = window as IntercomWindow
+const intercomWindow = (window as unknown) as IntercomWindow
```
Comment (nachomazzara) [inline src/components/Intercom/IntercomWidget.ts:5]: image

File: src/modules/transaction/utils.ts [hunk 3]
```diff
@@ -80,13 +82,13 @@ export function getEtherscanHref(
   return `${getEtherscanOrigin(network)}${pathname}`
 }
 
-export function getEtherscanOrigin(network: number = 1) {
-  switch (network) {
-    case 3:
+export function getEtherscanOrigin(chainId: number = ChainId.ETHEREUM_MAINNET) {
+  switch (chainId) {
```
Comment (nachomazzara) [inline src/modules/transaction/utils.ts:86]: Maybe we need to change the name of the name of the method `getEtherscanHref ` to `getTransactionLink` and `getEtherscanOrigin` to `getTransactionLinkOrigin` by adding


`https://explorer-mumbai.maticvigil.com` and `https://explorer-mainnet.maticvigil.com` for Matic

**Review by nachomazzara — APPROVED** [review]
LGTM

File: src/modules/transaction/txUtils.ts [hunk 4]
```diff
@@ -9,31 +13,30 @@ import {
...
-import { createEth } from '../../lib/eth'
 
 export async function getTransaction(
+  address: string,
+  chainId: ChainId,
   hash: string
 ): Promise<AnyTransaction | null> {
-  const eth = await createEth()
-  if (!eth) return null
+  const provider = await connection.createProvider(
```
Comment (nachomazzara) [inline src/modules/transaction/txUtils.ts:22]: Should we need to use `getProvider(chainId)` here?

File: src/lib/eth.ts [hunk 5]
```diff
@@ -18,19 +17,20 @@ export type EthereumWindow = Window & {
   ethereum?: Provider
 }
 
-export async function createEth() {
+export async function getProvider(
```
Comment (nachomazzara) [inline src/lib/eth.ts:20]: What do you think of splitting this method into two? 

`getConnectedProvider` which receives 0 params and return the current connected provider (OLD changed `getProvider()`. And `createProvider` or `getCustomProvider` or `createCustomProvider` or whatever name you fee comfortable which receives `chainId` and `providerType` and returns a provider created by dcl-connect: `connection.createProvider(providerType chainId)`

File: src/components/Intercom/IntercomWidget.ts (hunk 2 above)
Comment (nicosantangelo) [reply src/components/Intercom/IntercomWidget.ts:5]: JAJAJA

File: src/modules/transaction/utils.ts (hunk 3 above)
Comment (nicosantangelo) [reply src/modules/transaction/utils.ts:86]: hm you're right. We should change `EtherscanLink` too I think. I'll address both

File: src/modules/transaction/txUtils.ts (hunk 4 above)
Comment (nicosantangelo) [reply src/modules/transaction/txUtils.ts:22]: I wanted to be more specific and wasn't really sure about the new signature of `getProvider`. The new proposed implementation of `getConnectedProvider` makes this a lot more reliable. I'll change it :)

File: src/lib/eth.ts (hunk 5 above)
Comment (nicosantangelo) [reply src/lib/eth.ts:20]: love it, let me take a look at the implementation and report back if I see any problems

File: src/lib/chainConfiguration.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  [ChainId.ETHEREUM_MAINNET]:
+    'https://api.thegraph.com/subgraphs/name/decentraland/mana',
+  [ChainId.ETHEREUM_ROPSTEN]:
+    'https://api.thegraph.com/subgraphs/name/decentraland/mana-ropsten',
+  [ChainId.ETHEREUM_GOERLI]:
+    'https://api.thegraph.com/subgraphs/name/decentraland/mana-goerli',
+  [ChainId.MATIC_MAINNET]:
+    'https://api.staging.thegraph.com/subgraphs/name/decentraland/mana-matic',
+  [ChainId.MATIC_MUMBAI]:
+    'https://api.staging.thegraph.com/subgraphs/name/decentraland/mana-mumbai'
```
Comment (cazala) [inline src/lib/chainConfiguration.ts:14]: These are the updated subgraph urls:

```
Ethereum Mainnet: https://api.thegraph.com/subgraphs/name/decentraland/mana-ethereum-mainnet
Ethereum Ropsten: https://api.thegraph.com/subgraphs/name/decentraland/mana-ethereum-ropsten
Ethereum Goerli: https://api.thegraph.com/subgraphs/name/decentraland/mana-ethereum-goerli
Matic Mainnet: https://api.thegraph.com/subgraphs/name/decentraland/mana-matic-mainnet
Matic Mumbai: https://api.thegraph.com/subgraphs/name/decentraland/mana-matic-mumbai
```

**Review by cazala — CHANGES_REQUESTED** [review]

**Review by cazala — APPROVED** [review]

**Pull request merged**